### ISSUE TITLE
Question: on which Android versions is it needed and not needed?

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [drakeet/ToastCompat#15]
Was this issue that this library handles ever been fixed?
If so, from which version?

**Comment by danielgalion** [comment]
@AndroidDeveloperLB I could only tell you that recently I've seen in Crashlytics an issue with AlertDialog that was giving an error of BadTokenException. It happened on Samsung and Xiaomi devices with Android 9.
The project is built with pre-AndroidX components with targetSdkVersion 28.
I don't know either on which exactly Android versions and devices it happens.

**Comment by AndroidDeveloperLB** [comment]
You mean that it can occur on Android 9, right?
Do you know the cause of it in Android code? Maybe this way you can compare between what you know in Android 9 and 10.

I don't think it matters about android-x.
It's a toast. Plus it gets converted anyway to android-x.
And seeing that you used only the annotations of the support library, you can convert to Kotlin and not use anything of the support library (and android-x).

BTW, what's with the eyes emoji?

**Comment by danielgalion** [comment]
Yes, it seems it can occur because it has occured on Android 9. As I saw in Crashlytics of app deployed on Google Store.
I know only that it happened when the app was trying to call the show() method on the AlertDialog object. Unfortunately, nothing more.

I cannot reproduce the error in my Environment (Android 9 and 10 emulators and Samsung's smartphones with Android 5 and 6). I am just giving you (and maybe others) information that the issue happened. So, just in case you could also handle exceptions on the Android 9 version.

The eyes emoji means that I am also looking for the answer to your question.

**Comment by AndroidDeveloperLB** [comment]
How come it happens anyway? Is Google aware of the issue? Is there a report [here](https://issuetracker.google.com/)?
It occurs only for  Samsung and Xiaomi devices?

**Comment by AncientTree** [comment]
> How come it happens anyway? Is Google aware of the issue? Is there a report [here](https://issuetracker.google.com/)?
> It occurs only for Samsung and Xiaomi devices?

No, not only Samsung and Xiaomi, same on my OnePlus(witch run Android 9 now) and Android 6 emulator. 

Things seem very strange, that you can't believe:
``` java
@Override
public boolean onMenuItemClick(MenuItem item) {
        ...
        InputNotebookInfoDialog newbookDialog = new InputNotebookInfoDialog(MainActivity.this);
        newbookDialog.setOnConfirmClickListener(new onUpdateNotebookConfirm(bookChip.notebook));
        newbookDialog.setOnCancelClickListener(new onUpdateNotebookCanceled());
        newbookDialog.setInitShowNotebookInfo(bookChip.notebook);
        newbookDialog.show();
        ...
}
```
where `InputNotebookInfoDialog.setInitShowNotebookInfo` is look like:
```java
import android.app.Dialog;
public class InputNotebookInfoDialog extends Dialog {
    public EditText bookNameET, bookDescriptionET;
    ...
    public void setInitShowNotebookInfo(Notebook notebook){
        bookNameET.setText(notebook.Name);
        bookDescriptionET.setText(notebook.Description);
    }
}
```
This will report error, but
``` java
@Override
public boolean onMenuItemClick(MenuItem item) {
        ...
        InputNotebookInfoDialog newbookDialog = new InputNotebookInfoDialog(MainActivity.this);
        newbookDialog.setOnConfirmClickListener(new onUpdateNotebookConfirm(bookChip.notebook));
        newbookDialog.setOnCancelClickListener(new onUpdateNotebookCanceled());
        // newbookDialog.setInitShowNotebookInfo(bookChip.notebook);
        newbookDialog.show();
        ...
}
```
this is work fine. if it is advocated, I could show my full code. 
could this project resolve my poblem here?

**Comment by AndroidDeveloperLB** [comment]
@danielgalion Say, for some reason I can't star, watch, fork, create new issues here.
Was I banned for some reason? How can I check why, and say I'm sorry and request to un-ban me?
